test/functional/wallet_v3_txs.py

```diff
@@ -0,0 +1,560 @@
+        self.truc_tx_with_conflicting_sibling_change()
+        self.v3_tx_evicted_from_mempool_by_sibling()
+        self.v3_conflict_removed_from_mempool()
+        self.mempool_conflicts_removed_when_v3_conflict_removed()
+        self.max_tx_weight()
+        self.max_tx_child_weight()
+        self.user_input_weight_not_overwritten()
+        self.user_input_weight_not_overwritten_v3_child()
+        self.createpsbt_v3()
+        self.send_v3()
+        self.sendall_v3()
+        self.sendall_with_unconfirmed_v3()
+        self.walletcreatefundedpsbt_v3()
+
+    @cleanup
+    def tx_spends_unconfirmed_tx_with_wrong_version(self, version_a, version_b):
+        self.log.info(f"Test unavailable funds when v{version_a} tx spends unconfirmed v{version_b} tx")
+
+        self.generate(self.nodes[2], 1)
+
+        inputs=[]
+        outputs = {self.bob.getnewaddress() : 2.0}
+        self.send_tx(self.charlie, inputs, outputs, version_a)
+
+        assert_equal(self.bob.getbalances()["mine"]["trusted"], 0)
+        assert_greater_than(self.bob.getbalances()["mine"]["untrusted_pending"], 0)
+
+        inputs = []
+        outputs = {self.alice.getnewaddress() : 1.0}
+
+        raw_tx_v2 = self.bob.createrawtransaction(inputs=inputs, outputs=outputs, version=version_b)
+
+        assert_raises_rpc_error(
+            -4,
+            "Insufficient funds",
+            self.bob.fundrawtransaction,
+            raw_tx_v2, {'include_unsafe': True}
+        )
+
+    @cleanup
+    def v2_tx_spends_confirmed_v3_tx(self, version_a, version_b):
+        self.log.info(f"Test unavailable funds when v{version_a} tx spends confirmed v{version_b} tx")
+
+        self.generate(self.nodes[2], 1)
+
+        inputs=[]
+        outputs = {self.bob.getnewaddress() : 2.0}
+        self.send_tx(self.charlie, inputs, outputs, version_a)
+
+        assert_equal(self.bob.getbalances()["mine"]["trusted"], 0)
+        assert_greater_than(self.bob.getbalances()["mine"]["untrusted_pending"], 0)
+
+        inputs = []
+        outputs = {self.alice.getnewaddress() : 1.0}
+
+        self.generate(self.nodes[2], 1)
+
+        self.send_tx(self.bob, inputs, outputs, version_b)
+
+    @cleanup
+    def v3_utxos_appear_in_listunspent(self):
+        self.log.info("Test that unconfirmed v3 utxos still appear in listunspent")
+
+        inputs=[]
+        outputs = {self.alice.getnewaddress() : 2.0}
+        parent_txid = self.send_tx(self.charlie, inputs, outputs, 3)
+        assert_equal(self.alice.listunspent(minconf=0)[0]["txid"], parent_txid)
+
+    @cleanup
+    def truc_tx_with_conflicting_sibling(self):
+        # unconfirmed v3 tx to alice & bob
+        self.log.info("Test v3 transaction with conflicting sibling")
+        self.generate(self.nodes[2], 1)
+
+        inputs=[]
+        outputs = {self.alice.getnewaddress() : 2.0, self.bob.getnewaddress() : 2.0}
+        self.send_tx(self.charlie, inputs, outputs, 3)
+        parent_txid = self.charlie.getrawmempool()[0]
```

Comment:
In 09c9f828250e3840d7557374ff568204305e2679 "test: add truc wallet tests"

The `getrawmempool` can be avoided:

```suggestion
        parent_txid = self.send_tx(self.charlie, inputs, outputs, 3)
```